volumes/miovision/sql/create-mv-mio_cent.sql

```diff
@@ -0,0 +1,141 @@
+--3. manually investigated and fixed the centrelines that needed to be fixed (in the table)
+
+--1. Here is the code that finds centreline_ids that touch miovision intersections:
+CREATE TABLE miovision_api.centerline_miovision AS (
+
+    --find all the nodes in the centreline file that touch miovision intersections. You have to match with "from" and "to" nodes to get all the segments
+    WITH cent_int AS (
+        SELECT DISTINCT
+            ter.intersection_uid,
+            ter.lat,
+            ter.lng,
+            cent.centreline_id,
+            ci.latitude,
+            ci.longitude
+        FROM miovision_api.intersections AS ter
+        LEFT JOIN prj_volume.centreline AS cent ON ter.int_id = cent.to_intersection_id
+        LEFT JOIN prj_volume.centreline_intersections AS ci ON cent.from_intersection_id = ci.intersection_id
+        WHERE cent.feature_code_desc NOT IN ('Trail')
+
+        UNION ALL
+
+        SELECT DISTINCT
+            ter.intersection_uid,
+            ter.lat,
+            ter.lng,
+            cent.centreline_id,
+            ci.latitude,
+            ci.longitude
+        FROM miovision_api.intersections AS ter
+        LEFT JOIN prj_volume.centreline AS cent ON ter.int_id = cent.from_intersection_id
+        LEFT JOIN prj_volume.centreline_intersections AS ci ON cent.to_intersection_id = ci.intersection_id
+        WHERE cent.feature_code_desc NOT IN ('Trail')
+    ),
+
+    --determine difference in the "non-miovision node" co-ordinates - figure out if we're dealing with the north-south or east-west street in the intersection
+    ll_diff AS (
+        SELECT
+            ci.intersection_uid,
+            ci.lat,
+            ci.lng,
+            ci.centreline_id,
+            ci.latitude,
+            ci.longitude,
+            ci.latitude - ci.lat AS lat_diff,
+            ci.longitude - ci.lng AS lng_diff,
+            CASE
+                WHEN abs(ci.latitude - ci.lat) > abs(ci.longitude - ci.lng) THEN 'NS'
+                WHEN abs(ci.longitude - ci.lng) > abs(ci.latitude - ci.lat) THEN 'EW'
+            END AS nsew
+        FROM cent_int AS ci
+    ),
+
+    --determine which segments are N, S, E, W so that we can match with miovision leg + intersection_uid
+    cent_leg AS (
+        SELECT
+            ld.intersection_uid,
+            ld.lat,
+            ld.lng,
+            ld.centreline_id,
+            ld.latitude,
+            ld.longitude,
+            ld.nsew,
+            CASE
+                WHEN ld.nsew = 'NS' AND ld.latitude > ld.lat THEN 'N'
+                WHEN ld.nsew = 'NS' AND ld.latitude < ld.lat THEN 'S'
+                WHEN ld.nsew = 'EW' AND ld.longitude > ld.lng THEN 'E'
+                WHEN ld.nsew = 'EW' AND ld.longitude < ld.lng THEN 'W'
+            END AS leg
+        FROM ll_diff AS ld
+    )
+
+    --yeah there's a ton of stuff we needed for calculations in those earlier tables but now we just need:
+    SELECT
```

Comment:
we can do this select in the cent_leg cte directly 😎

Reply:
Ooh I saw that! Implemented!!!